lib/active_merchant/billing/gateway.rb

```diff
@@ -193,6 +193,10 @@ def scrub(transcript)
         raise RuntimeError.new("This gateway does not support scrubbing.")
       end
 
+      def network_tokenization_supported
```

Comment:
Why have `network_tokenization_supported` public and have it on `Gateway`? It's only ever used (for now) by Braintree Blue; by making it public, we're acknowledging this is now part of the public API. We're still not sure what this logic should look like, which means it will probably change in the near future. Having it public means dealing with deprecation and breaking changes as soon as you release the gem. I'd keep it as a hidden implementation detail in `BraintreeBlue`.